File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/model/FilteredView.kt
package com.storysphere.storyreader.model

data class FilteredView(
    val id: String,
    val userId: String,
    val name: String,
    val description: String? = null,
    val query: FilterQuery,
    val order: Int = 0,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis(),
    val syncStatus: SyncStatus = SyncStatus.SYNCED,
    val lastSyncedAt: Long? = null
)

data class FilterQuery(
    val tags: List<String>? = null,
    val authors: List<String>? = null,
    val genres: List<String>? = null,
    val completionStatus: CompletionStatus? = null,
    val dateRange: DateRange? = null,
    val searchText: String? = null
)

enum class CompletionStatus {
    ALL,
    COMPLETED,
    ONGOING,
    HIATUS
}

data class DateRange(
    val startDate: Long? = null,
    val endDate: Long? = null
)

File: Novels/packages/5-mobile-ios/StoryReader/Model/FilteredView.swift
import Foundation

// FilteredView model - Dynamic query views
struct FilteredView: Codable, Identifiable {
    let id: String
    let userId: String
    let name: String
    let description: String?
    let query: FilterQuery
    let isAutoUpdating: Bool
    let displayOrder: Int
    let createdAt: Date
    let updatedAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case name
        case description
        case query
        case isAutoUpdating = "is_auto_updating"
        case displayOrder = "display_order"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

